myia/simplify_types.py

```diff
@@ -303,6 +303,12 @@ def to_canonical(self, arg, orig_t: AbstractTuple, coerce):
     return tuple(self(x, o, coerce) for x, o in zip(arg, oe))
 
 
+@overload  # noqa: F811
+def to_canonical(self, arg, orig_t: AbstractRandomState, coerce):
+    # arg must be a RandomStateWrapper
+    return arg
```

Comment:
You need to check that arg actually is a RandomStateWrapper. Once a function is compiled for the first time, `to_canonical` is used to validate the arguments.

Reply:
Done!